### PR TITLE
fix: remove invalid multi-tenant VPE references

### PR DESCRIPTION
**Pull request by vburckhardt**
### Description

fixes https://github.com/terraform-ibm-modules/terraform-ibm-vpe-gateway/issues/391
fixes https://github.com/terraform-ibm-modules/terraform-ibm-vpe-gateway/issues/392

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [x] Major release (`X.x.x`)

##### Release notes content

fix: remove invalid multi-tenant VPE references
fix: ensure idempotent of existing VPEs

BREAKING CHANGE: all existing VPE created in previous version of this module will be deleted and immediately re-created. The IP of the VPE may change as part of this process.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

**Comment by vburckhardt** [comment]
note: does not cover https://github.com/terraform-ibm-modules/terraform-ibm-vpe-gateway/issues/394 which will need to be done separately

**Review by vbontempi — COMMENTED** [review]
PR looks ok.
Just a little question to be sure: @vburckhardt the `every-mt-type` test is not executed with PR ci tests, is this correct?

**Review by shemau — APPROVED** [review]
This looks good.

I do not understand why certain cloud_services are being removed and if this leads to work to add databases and secrets-manager back.

File: variables.tf [hunk 1]
```diff
@@ -73,12 +73,6 @@ variable "cloud_services" {
         "cloud-object-storage",
         "codeengine",
         "container-registry",
-        "databases-for-cassandra",
-        "databases-for-elasticsearch",
-        "databases-for-enterprisedb",
-        "databases-for-mongodb",
-        "databases-for-postgresql",
-        "databases-for-redis",
```
Comment (shemau) [inline variables.tf:81]: databases is included the list in #396 at https://cloud.ibm.com/docs/vpc?topic=vpc-vpe-supported-services

File: variables.tf [hunk 2]
```diff
@@ -87,12 +81,10 @@ variable "cloud_services" {
         "hs-crypto",
         "hyperp-dbaas-mongodb",
         "hyperp-dbaas-postgresql",
-        "iam-identity",
         "iam-svcs",
         "is",
         "kms",
         "resource-controller",
-        "secrets-manager",
```
Comment (shemau) [inline variables.tf:95]: secrets-manager is included the list in #396 at https://cloud.ibm.com/docs/vpc?topic=vpc-vpe-supported-services

**Comment by shemau** [comment]
> @shemau - Yes, those should not have been there as, cloud_services variable should only take multi-tenant endpoint. ICD and sm are single tenant (pointing to a specific instance), and as such should be targeted through cloud_service_by_crn
> 
> @vbontempi - I had captured that need separately at #394

Thanks. For the record, coming soon, the ICD postgresql example will cover VPE wrt to the single tenant instance that is created.

**Pull request merged**